### PR TITLE
Make real output and ref output in the same type

### PR DESCRIPTION
**Pull request by qkmiao**
C++ code:
https://android.googlesource.com/platform/external/deqp/+/master/modules/gles3/functional/es3fShaderCommonFunctionTests.cpp#978

https://android.googlesource.com/platform/external/deqp/+/master/modules/gles3/functional/es3fShaderCommonFunctionTests.cpp#1050

### CODE REVIEW

File: sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js [hunk 1]
```diff
@@ -1051,6 +1051,12 @@ goog.scope(function() {
         for (var compNdx = 0; compNdx < scalarSize; compNdx++) {
             in0 = inputs[0][compNdx];
             out0 = outputs[0][compNdx];
+
+            // Convert int to uint because ref out is in uint format.
+            var view = new DataView(new ArrayBuffer(4));
+            view.setUint32(0, out0, true);
```
Comment (zhenyao) [inline sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js]: Shouldn't this be setInt32() instead?

File: sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js [hunk 2]
```diff
@@ -1051,6 +1051,12 @@ goog.scope(function() {
...
+            out0 = view.getUint32(0, true);
+
             refOut0 = tcuFloat.newFloat32(in0).bits();
             ulpDiff = Math.abs(Math.floor(out0) - Math.floor(refOut0));
```
Comment (zhenyao) [inline sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js]: No need to call floor() on out0 after the above conversion, right?

File: sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js [hunk 3]
```diff
@@ -1144,9 +1150,13 @@ goog.scope(function() {
         for (var compNdx = 0; compNdx < scalarSize; compNdx++) {
             in0 = inputs[0][compNdx];
             out0 = outputs[0][compNdx];
-            refOut0 = tcuFloat.newFloat32(in0).bits();
-            ulpDiff = Math.abs(Math.floor(in0) - Math.floor(out0));
 
+            // Convert int to float
+            var view = new DataView(new ArrayBuffer(4));
+            view.setUint32(0, in0, true);
```
Comment (zhenyao) [inline sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js]: Same here, shouldn't this be setInt32() instead?

Comment (zhenyao) [inline sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js]: The c++ code did this conversion to also out0.

File: sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js (hunk 1 above)
Comment (qkmiao) [inline sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js]: Also setInt32() can be used. getUint32() is critical.

File: sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js (hunk 2 above)
Comment (qkmiao) [inline sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js]: Yes. out0 and refOut0 are both unsigned int.

File: sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js (hunk 3 above)
Comment (qkmiao) [inline sdk/tests/deqp/functional/gles3/es3fShaderCommonFunctionTests.js]: out0 is alread float type. No data type change for "const float		out0		= ((const float*)outputs[0])[compNdx];" While for in0 it is interpreted as float from int.

**Comment by zhenyao** [comment]
thanks. lgtm

**Pull request merged**